docs/advanced-functionality/drag-drop-csv-tsv.rst

```diff
@@ -0,0 +1,106 @@
+Adding extra metadata via CSV/TSV/XLSX
+======================================
+
+A common use case is to have additional metadata which you would like to add to the current dataset. If you created the dataset itself, then you may wish to keep certain data out of the dataset, as it may change frequently or be sensitive information which you don't want to share publicly.
+
+Additional metadata (CSV / TSV / XLSX file(s)) can be dragged onto an existing dataset in Auspice. These extra data are processed within the browser, so no information leaves the client, which can be useful for viewing private metadata.
+
+The general format is compatible with other popular tools such as `MicroReact <https://microreact.org/>`__. The first column defines the names of the strains / samples in the tree, while the first row (header row) defines the metadata names. You can add as many columns you want, each will result in a different colouring of the data being made available. The separator can be either a tab character or a comma & the file extension should be ``.tsv`` or ``.csv``, respectively. Excel files with file extension ``.xlsx`` are also supported, but the metadata must be in the first sheet of the workbook. Older Excel files with the ``.xls`` extension are not supported.
+
+Example:
+--------
+
+A TSV file as follows can be dragged onto `nextstrain.org/zika <https://nextstrain.org/zika>`__ to add a "secret" color-by:
+
+.. code:: text
+
+   strain  secret
+   USVI/19/2016    A
+   USVI/28/2016    B
+   USVI/41/2016    C
+   USVI/42/2016    C
+
+.. figure:: ../assets/csv-extra-data.png
+   :alt: auspice with extra data shown via csv
+
+   auspice with extra data shown via csv
+
+A more complex metadata file may look like the following, which makes use of additional features available. This defines colours for the metadata (e.g. ``A`` is yellow, ``B`` is orange) as well as associating strains with (made up) geographic coordinates.
+
+.. code:: text
```

Comment:
Would be amazing if rst were able to line up the columns but I don't think this is possible (and I wouldn't want to use a table to achieve this)

Reply:
I also don't think it's possible, since the intent of the code block is to display text literally.

It looks like there's a way to [set the tab width globally](https://docutils.sourceforge.io/docs/user/config.html#tab-width), but that would affect all code blocks across the docs project.

Maybe it could be displayed in a table with a button to copy the "raw TSV" to clipboard, but that seems hacky.

Reply:
I thought this would be the case - thanks for looking into it!

Reply:
https://github.com/nextstrain/auspice/pull/1638